baseapp/baseapp.go

```diff
@@ -790,3 +847,27 @@ func createEvents(msg sdk.Msg) sdk.Events {
 
 	return sdk.Events{msgEvent}
 }
+
+// DefaultProcessProposal returns the default implementation for processing an ABCI proposal.
+// Every transaction in the proposal must pass 2 conditions:
+//
+// 1. The transaction bytes must decode to a valid transaction.
+// 2. The transaction must be valid (i.e. pass runTx)
+//
+// If any transaction fails to pass either condition, the proposal is rejected.
+func (app *BaseApp) DefaultProcessProposal() sdk.ProcessProposalHandler {
+	return func(ctx sdk.Context, req abci.RequestProcessProposal) abci.ResponseProcessProposal {
```

Comment:
Food for thought: As an initial implementation for ABCI 1.0 why not just always return `abci.ResponseProcessProposal_ACCEPT`?

Some benefits:
* It gives applications the exact same behavior that exists in the SDK today, therefore reduces potential for any liveness risks or bugs.
* It's simpler and easier for you to test.
* It's more performant.

My understanding is that the introduction of `ProcessProposal` was not intended specifically to solve some existing consensus issue on Cosmos SDK chains, rather it was introduced to give application developers flexibility (like us at dYdX) to reject proposals. So even if your default implementation _always_ returns `abci.ResponseProcessProposal_ACCEPT`, you are still fulfilling the promise of `ProcessProposal` by allowing application developers to optionally implement their own logic via `SetProcessProposal`.

If in the future if there emerges a strong case to return `REJECT` in `DefaultProcessProposal` (i.e. if you want to validate that the transactions in a block do not exceed `MaxGas`, or something like that), then you could easily introduce such changes iteratively in a future version of the SDK.

Running every transaction through `CheckTx` seems a bit extreme to me and it's not clear that it solves any specific problem that exists in any existing chain today, however I could be ignorant or wrong about this. It also just actively makes upgrades more risky. For example, is it possible that some existing chain could have added some custom AnteHandler which doesn't necessarily expect to be invoked during `ProcessProposal`? Is this something that developers upgrading to `v0.47` be aware of and audit in their codebases when upgrading?

In any case, dYdX be using `SetProcessProposal` so the default behavior is of no concern to us. This is merely my own personal opinion on a more iterative upgrade path for the SDK that reduces some of the surface area while still retaining the most important feature (the ability for developers to customize `ProcessProposal` via `SetProcessProposal`).

Reply:
Thanks @prettymuchbryce you make some good points.  I've captured them in https://github.com/cosmos/cosmos-sdk/issues/13810 for further review and implementation.  If we decide it's something we want I @JeancarloBarrios or I can knock it out pretty quickly.  At this moment to change direction would mean a thoughtful rewrite.  Perhaps it's warranted but this PR was opened Oct 5 so we feel this iteration is good enough for an MVP.  Let's continue the discussion! :pray:

Reply:
That makes sense. Thanks for humoring me! 😄